src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -983,10 +983,11 @@ private static BoundExpression NullableAlwaysHasValue(BoundExpression expression
             TypeSymbol rewrittenType)
         {
             Debug.Assert((object)conversion.Method != null && !conversion.Method.ReturnsVoid && conversion.Method.ParameterCount == 1);
-            if (rewrittenOperand.Type != conversion.Method.ParameterTypes[0])
+            if (rewrittenOperand.Type.IsNullableType() &&
```

Comment:
Do we need to consider the possibility of a `null` value for `rewrittenOperand.Type` here?